src/notation/view/internal/undoredomodel.cpp

```diff
@@ -67,6 +67,10 @@ void UndoRedoModel::load()
         });
     });
 
+    context()->currentProjectChanged().onNotify(this, [this]() {
+        load();
```

Comment:
I think we can simply call `emit stackChanged();` here. No need to re-subscribe to notifications, if I'm not mistaken.